File: src/com/train/algorithm/tree/implementInC++/MinimumDepthofBinaryTree.cpp
//
// Created by 罗斯 on 21/10/2022.
//
#include <algorithm>
using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    int minDepth(TreeNode* root) {
        if (!root) return 0;
        if (!root->left && !root->right) return 1;
        int l = root->left ? minDepth(root->left) : INT_MAX;
        int r = root->right ? minDepth(root->right) : INT_MAX;
        return min(l, r) + 1;
    }
};

File: src/com/train/algorithm/tree/implementInPython/MinimumDepthOfBinaryTree.py
class TreeNode(object):
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution(object):
    def minDepth(self, root):
        if not root: return 0
        if not root.left and not root.right: return 1
        l = self.minDepth(root.left)
        r = self.minDepth(root.right)
        if not root.left: return r + 1
        if not root.right: return l + 1
        return 1 + min(l, r)